src/osd/OSD.cc

```diff
@@ -4287,6 +4304,51 @@ int OSD::shutdown()
     cct->_conf.apply_changes(nullptr);
   }
 
+  if (cct->_conf->osd_fast_shutdown) {
+    // first, stop new task from being taken from op_shardedwq
+    op_shardedwq.stop_for_fast_shutdown();
+
+    // do we still need to drain ???
```

Comment:
not when we just discarded the queue!

Reply:
the requests that are in flight are being processed by osd_op_tp, which you drain below, so they should be handled already. in fact, wq->drain() just calls tp->drain()